File: VB-v101/SqRoot101.vb
Public Class SqRoot101
    '15/9/21 Nev Goodyer: created from pseudo at https://bwattle.github.io/SquareRoot/JS/Q23.html
    Private Sub BtnRoot_Click(sender As Object, e As EventArgs) Handles btnCalc.Click
        Dim number As Single
        Dim decPlaces As Int16
        Dim minimum As Single
        Dim maximum As Single
        Dim middle As Single
        Dim accuracy As Single

        number = txtNo1.Text
        decPlaces = txtNo2.Text
        minimum = 0
        maximum = number
        middle = (minimum + maximum) / 2
        accuracy = 10 ^ (-1 * decPlaces)
        While Math.Abs((middle * middle) - number) > accuracy
            If middle * middle > number Then
                maximum = middle
            Else
                minimum = middle
            End If
            middle = (minimum + maximum) / 2
        End While
        txtResult.Text = Int(middle / accuracy) * accuracy
        txtIterations.Text = "The square root of " & number & " correct to " & decPlaces & " decimal places is " & middle
    End Sub
End Class

File: Py/SqRoot-101.py
from ast import literal_eval
# 18/9/21 Nev Goodyer - basic working version - rounding as in question
number = float(input("Enter the number you want the square root of: "))
decPlaces = int(input("Enter the number of decimal places of accuracy: "))
minimum = 0
maximum = number
middle = (minimum + maximum) / 2
accuracy = 10 ** (-1 * decPlaces)
while abs((middle * middle) - number) > accuracy:
    if middle * middle > number:
        maximum = middle
    else:
        minimum = middle
    middle = (minimum + maximum) / 2
middle = int(middle / accuracy) * accuracy  # accuracy rounding as in question
#middle = round(middle,decPlaces)
print(middle)
print("The square root of ", number, " to ", decPlaces, " decimal places is ", middle)
